packages/lexical/src/LexicalSelection.ts

```diff
@@ -1703,7 +1703,12 @@ export class RangeSelection implements BaseSelection {
       if (this.forwardDeletion(anchor, anchorNode, isBackward)) {
         return;
       }
-
+      if ($isElementNode(anchorNode)) {
+        const childNode = anchorNode.getChildAtIndex(0);
+        if ($isDecoratorNode(childNode) && anchor.offset === 0) {
+          return;
+        }
+      }
```

Comment:
I'd probably re-order this to short-circuit with fewer function calls, e.g.

```suggestion
      if (anchor.offset === 0 && $isElementNode(anchorNode) && $isDecoratorNode(anchorNode.getFirstChild())) {
        return;
      }
```